src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -831,6 +833,12 @@ await UpdateTopologyAsync(new UpdateTopologyParameters(serverNode)
                         ApplicationIdentifier = applicationIdentifier
                     }).ConfigureAwait(false);
 
+                    if (Disposed && _nodeSelector == null)
```

Comment:
you cannot since if the `UpdateTopologyAsync` return false, it might be for different reasons (like `lockTaken == false` or  `_disableTopologyUpdates`) and we only care in case of Dispose